.github/workflows/native.yml

```diff
@@ -0,0 +1,49 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        config: [ Debug, Release ]
+
+    steps:
+    - name: prerequisites
+      run: |
+        echo 'deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic main' | sudo tee -a /etc/apt/sources.list
+        sudo apt update
+        sudo apt install -y ninja-build p7zip clang lld
+        sudo update-alternatives --install /usr/bin/cc  cc  /usr/bin/clang 200
+        sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 200
+        sudo update-alternatives --install /usr/bin/ld  ld  /usr/bin/lld 200
+    - name: clone
+      run: |
+        git clone --depth 10000 -b master https://github.com/llvm/llvm-project.git .
+        git checkout 506992848
+        mkdir build
+        git clone https://github.com/$GITHUB_REPOSITORY llvm/tools/llvm-mctoll
+        cd llvm/tools/llvm-mctoll && git checkout $GITHUB_SHA
+    - run: |
+        echo "::add-matcher::$GITHUB_WORKSPACE/llvm/tools/llvm-mctoll/.github/cpp.json"
+    - name: cmake
+      working-directory: build
+      run: cmake -G Ninja -DLLVM_INSTALL_TOOLCHAIN_ONLY=ON -DLLVM_TARGETS_TO_BUILD="X86;ARM" -DLLVM_ENABLE_PROJECTS=clang -DLLVM_ENABLE_ASSERTIONS=ON -DCMAKE_BUILD_TYPE=${{ matrix.config }} -DLLVM_ENABLE_MODULES=ON ../llvm
+    - working-directory: build
+      run: ninja llvm-mctoll opt clang lli
```

Comment:
I was puzzling over the reason for the test failures. From my brief encounters with Github actions, it appears that there may be "persistence" related issues between jobs. 

So, as an experiment, may be you can just run `ninja check-mctoll` right here instead of `ninja llvm-mctoll opt clang lli`? 

I am assuming that you split the targets to ease the high memory usage during parallel builds - particularly while building `clang` and `opt` targets. If so, may be you can run `ninja clang && ninja opt && ninja check-mctoll`.

Reply:
No that's just to keep the check step clean (focused on the actual tests).